dartsim/src/EntityManagementFeatures.cc

```diff
@@ -436,25 +442,37 @@ Identity EntityManagementFeatures::GetJoint(
 Identity EntityManagementFeatures::GetJoint(
     const Identity &_modelID, const std::string &_jointName) const
 {
-  DartJoint *const joint =
-      this->ReferenceInterface<ModelInfo>(_modelID)->model->getJoint(
-          _jointName);
+  /// \todo(srmainwaring) remove debug code
+  // std::cerr << this->DebugModels();
+  // std::cerr << this->DebugLinks();
+  // std::cerr << this->DebugJoints();
 
-  // If the joint doesn't exist in "joints", it means the containing entity has
-  // been removed.
-  if (this->joints.HasEntity(joint))
-  {
-    const std::size_t jointID = this->joints.IdentityOf(joint);
-    return this->GenerateIdentity(jointID, this->joints.at(jointID));
-  }
-  else
+  const auto &modelInfo = this->ReferenceInterface<ModelInfo>(_modelID);
+  for (const auto &jointInfo : modelInfo->joints)
```

Comment:
Now that we have `jointsByName`,  can we just lookup the joint's name in there instead of having a for loop here?

Reply:
Updated in 9553dea7 (using the internally generated fully scoped joint name).